File: example/cpp/test_log_download_api.cpp
/**
 * test_log_download_api.cpp
 * @brief 日志下载相关接口
 * @attention 确保机械臂供电正常、网络通信正常，且日志保存路径有效
 * @note 运行步骤
 *       编译: cd build && cmake .. && make
 *       运行: ./test_log_download_api
 */

#include <iostream>
#include "cpp/interface/tl_api.h"

int main() {
    // 定义IP地址和端口
    std::string ip = "192.168.1.13";
    std::string port = "6001";

    // 定义日志文件路径
    int count = 10;
    std::string log_path = "./log";
    
    // 获取 SDK 版本
    std::string version = get_library_version();
    std::cout << "[INFO] SDK 版本: " << version << std::endl;

    // 连接控制器
    std::cout << "[INFO] 连接控制器..." << std::endl;
    auto socket_fd = connect_robot(ip, port);
    if (socket_fd < 0) {
        std::cerr << "[ERROR] 连接失败，程序退出!"<< std::endl;
        return -1;
    }
    std::cout << "[INFO] 连接成功 (socket=" << socket_fd << ")" << std::endl;

    // 下载日志
    std::cout << "[INFO] 开始下载日志..." << std::endl;
    auto ret = log_download_by_quantity(socket_fd, count, log_path);
    if (ret != Result::SUCCESS) {
        std::cerr << "[ERROR] 日志下载失败，程序退出!" << std::endl;
        return -1;
    }
    std::cout << "[INFO] 日志下载完成" << std::endl;

    // 断开控制器连接
    std::cout << "[INFO] 断开控制器连接..." << std::endl;
    ret = disconnect_robot(socket_fd);
    if (ret != Result::SUCCESS) {
        std::cerr << "[ERROR] 断开连接失败，程序退出!"<< std::endl;
        return -1;
    }
    std::cout << "[INFO] 断开连接成功" << std::endl;

    return 0;
}

File: example/py/test_log_download_api.py
#!/usr/bin/env python3
#
# test_log_download_api.py
# @brief 日志下载相关接口
# @attention 确保机械臂供电正常、网络通信正常，且日志保存路径有效
# @note 运行步骤
#       运行: python3 test_log_download_api.py
#

import sys
import time

# 添加 SDK Python 绑定路径
sys.path.append("../../lib/linux/x86/python")
from tl_interface import *


def main():
    # 定义IP地址和端口
    ip = "192.168.1.13"
    port = "6001"

    # 定义日志文件路径
    count = 10
    log_path = "./log"

    # 获取 SDK 版本
    version = get_library_version()
    print("[INFO] SDK 版本: " + version)

    # 连接控制器
    print("[INFO] 连接控制器...")
    socket_fd = connect_robot(ip, port)
    if socket_fd < 0:
        print("[ERROR] 连接失败，程序退出!")
        return -1
    print("[INFO] 连接成功 (socket=" + str(socket_fd) + ")")

    # 下载日志
    print("[INFO] 开始下载日志...")
    ret = log_download_by_quantity(socket_fd, count, log_path)
    if ret != SUCCESS:
        print("[ERROR] 日志下载失败，程序退出!")
        return -1
    print("[INFO] 日志下载完成")

    # 断开控制器连接
    print("[INFO] 断开控制器连接...")
    ret = disconnect_robot(socket_fd)
    if ret != SUCCESS:
        print("[ERROR] 断开连接失败，程序退出!")
        return -1
    print("[INFO] 断开连接成功")

    return 0


if __name__ == "__main__":
    exit(main())
